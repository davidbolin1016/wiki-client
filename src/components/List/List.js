import React from 'react';
import { Link } from 'react-router-dom';
import UserContext from '../../user-context/UserContext';

export default class List extends React.Component {

  static contextType = UserContext;
  
  render() {
    const listElements = this.props.pageList.map((ele, i) => {
      return (
        <li className="listed-page" key={ele.id}>
          <Link to={'/pages/' + ele.id}>{ele.page_name}</Link>
          <div className="edit-or-delete">
            <button disabled={this.context.homepage === `/pages/${ele.id}`} onClick={() => this.props.handleDelete(ele.id)}>Delete</button>
            <button onClick={() => this.props.handleEdit(ele.id)}> Edit </button>
          </div>
        </li>
      );
    });
  
    return (
      <ul className="page-list">
        {listElements}
      </ul>
    );
  }
}
  
  