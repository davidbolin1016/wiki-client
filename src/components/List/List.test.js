import React from 'react';
import ReactDOM from 'react-dom';
import {BrowserRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import List from './List';

const testList = [
  {
      "id": 69,
      "page_name": "New Page Thu May 02 2019 10:10:03 GMT-0700 (Pacific Daylight Time)",
      "date_created": "2019-05-02T17:10:11.468Z",
      "date_modified": "2019-05-02T17:10:11.468Z"
  },
  {
      "id": 71,
      "page_name": "New Page Thu May 02 2019 11:44:48 GMT-0700 (Pacific Daylight Time)",
      "date_created": "2019-05-02T18:44:57.131Z",
      "date_modified": "2019-05-02T18:44:57.131Z"
  },
  {
      "id": 70,
      "page_name": "Testing internal linking",
      "date_created": "2019-05-02T18:04:01.893Z",
      "date_modified": "2019-05-02T18:49:52.392Z"
  }
];

describe('List component', ()=> {
  const handleDelete = jest.fn();
  const handleEdit = jest.fn();

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <List pageList={[]} handleDelete={handleDelete} handleEdit={handleEdit} />
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders as expected when provided with a list of pages', () => {
    const tree = renderer
      .create(<BrowserRouter>
        <List pageList={testList} handleDelete={handleDelete} handleEdit={handleEdit} />
      </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot(); 
  });
  
})
