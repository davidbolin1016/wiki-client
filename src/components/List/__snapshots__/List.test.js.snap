// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders as expected when provided with a list of pages 1`] = `
<ul
  className="page-list"
>
  <li
    className="listed-page"
  >
    <a
      href="/pages/69"
      onClick={[Function]}
    >
      New Page Thu May 02 2019 10:10:03 GMT-0700 (Pacific Daylight Time)
    </a>
    <div
      className="edit-or-delete"
    >
      <button
        disabled={false}
        onClick={[Function]}
      >
        Delete
      </button>
      <button
        onClick={[Function]}
      >
         Edit 
      </button>
    </div>
  </li>
  <li
    className="listed-page"
  >
    <a
      href="/pages/71"
      onClick={[Function]}
    >
      New Page Thu May 02 2019 11:44:48 GMT-0700 (Pacific Daylight Time)
    </a>
    <div
      className="edit-or-delete"
    >
      <button
        disabled={false}
        onClick={[Function]}
      >
        Delete
      </button>
      <button
        onClick={[Function]}
      >
         Edit 
      </button>
    </div>
  </li>
  <li
    className="listed-page"
  >
    <a
      href="/pages/70"
      onClick={[Function]}
    >
      Testing internal linking
    </a>
    <div
      className="edit-or-delete"
    >
      <button
        disabled={false}
        onClick={[Function]}
      >
        Delete
      </button>
      <button
        onClick={[Function]}
      >
         Edit 
      </button>
    </div>
  </li>
</ul>
`;
